### DISCUSSION TITLE
How do I add arrowhead for my edges?

### DISCUSSION THREAD

**Discussion by dinkoivanov-gepard** [eclipse-glsp/glsp#1113] [Q&A] [answered]
Hi,

First of all, great project, thanks for your work!
I've been a long time user of GEF in Eclipse, been sceptical about diagramming in web, but I've recently found GLSP and it rocks!

Now to the question - I'm doing a quick prototype based on the node-json-vscode template (https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/node-json-vscode) and I've added support for edges between tasks. I'm using this code to create my edges:

```
GEdge.builder()
            .id(`${sourceStep}:${targetStep}`)
            .sourceId(sourceStep)
            .targetId(targetStep)
            .build();
```

How do I add an arrow head to point to the target?

Thanks,
Dinko

**Comment by CamilleLetavernier** [comment, marked as answer]
Hi Dinko,

On the server, you need to specify a `type` attribute for your Edge, and then, on your client-side Diagram Module, map this `type` to an EdgeView that supports Arrow decorations. This is typically done with a subclass of `PolylineEdgeView`, which implements the `renderAdditionals` method to add an SVG triangle at the end of the last segment.

You can find examples of arrow decoration in these discussions:

https://github.com/eclipse-glsp/glsp/discussions/1058#discussioncomment-6497529
https://github.com/eclipse-glsp/glsp/discussions/753#discussion-4431786

**Reply by dinkoivanov-gepard** [reply]
Thanks @CamilleLetavernier !

**Reply by jdeantoni** [reply]
Hi ! 

Sorry to revive this thread, but I tried the linked solution and encountered a strange error:
![image](https://github.com/user-attachments/assets/2717e9bf-7502-4c5b-8fb4-5bfc88452dc1)

here is my view:
`
@injectable()
export class CustomEdgeView extends PolylineEdgeView {
  protected override renderAdditionals(edge: GEdge, segments: Point[], context: RenderingContext): VNode[] {
    JSX; //useless but avoid configuring eslint
    const additionals = super.renderAdditionals(edge, segments, context);
    const endP1 = segments[segments.length - 2];
    const endP2 = segments[segments.length - 1];

    // arrow depends on the type of the BPMNEdge
      const headArrow: any = (
          <path
              class-sprotty-edge={true}
              class-arrow={true}
              d='M 1,0 L 14,-4 L 14,4 Z'
              transform={`rotate(${toDegrees(angleOfPoint({ x: endP1.x - endP2.x, y: endP1.y - endP2.y }))}
              ${endP2.x} ${endP2.y}) translate(${endP2.x} ${endP2.y}
          )`}
          />
      );
    additionals.push(headArrow);
    return additionals;
  }
}`

any idea of where it could come from ?
